core/src/main/java/com/linecorp/armeria/server/http/HttpServerPipelineConfigurator.java

```diff
@@ -91,6 +91,7 @@ public HttpServerPipelineConfigurator(
     @Override
     protected void initChannel(Channel ch) throws Exception {
         final ChannelPipeline p = ch.pipeline();
+        p.addLast(config.maxConnectionHandler());
```

Comment:
How about adding the handler to the `ServerChannel` pipeline? You can use `ServerBootstrap.handler()` to add a `ChannelInboundHandler` whose `channelRead()` message is a newly accepted `Channel` instance. You can close the connection even before the pipeline is initialized. For more information, please read `ServerBootstrap.ServerBootstrapAcceptor`.